spring-integration-core/src/main/java/org/springframework/integration/support/management/micrometer/MicrometerMetricsCaptor.java

```diff
@@ -49,31 +50,44 @@
 
 	public static final String MICROMETER_CAPTOR_NAME = "integrationMicrometerMetricsCaptor";
 
-	protected final MeterRegistry meterRegistry; // NOSONAR
+	private MeterRegistry meterRegistry;
+
+	private ObjectProvider<MeterRegistry> meterRegistryProvider;
 
 	public MicrometerMetricsCaptor(MeterRegistry meterRegistry) {
 		Assert.notNull(meterRegistry, "meterRegistry cannot be null");
 		this.meterRegistry = meterRegistry;
 	}
 
+	MicrometerMetricsCaptor(ObjectProvider<MeterRegistry> meterRegistryProvider) {
+		this.meterRegistryProvider = meterRegistryProvider;
+	}
+
+	public MeterRegistry getMeterRegistry() {
+		if (this.meterRegistry == null) {
+			this.meterRegistry = this.meterRegistryProvider.getIfUnique();
```

Comment:
Shouldn't we get the first if there are more than one? (`provider.stream().findFirst()`) This would then be consistent with the existing code.

Reply:
Well, it is better to use a `getIfUnique()` since this one returns for a `@Primary` one - exactly what we want to have in case of Spring Boot.
Apparently the current solution is still bug, since we inject really only one `MeterRegistry` (Prometheus), but not the composite built by Boot including all the rest available registries, e.g. WaveFront one.

Reply:
OK; just waiting for travis, then.